MzML/Mzml.cs

```diff
@@ -477,11 +477,32 @@ private static MsDataScan GetMsDataOneBasedScanFromConnection(Generated.mzMLType
             }
 
             DissociationType dissociationType = DissociationType.Unknown;
+
             if (_mzMLConnection.run.spectrumList.spectrum[oneBasedIndex - 1].precursorList.precursor[0].activation.cvParam != null)
             {
+                // for EThcD scans, the dissociation type will not be listed as EThcD. it will be 2 different dissociation types
+                // in the list, one as ETD and one with HCD. so we need to check for that case and interpret it as EThcD.
+                List<DissociationType> scanDissociationTypes = new List<DissociationType>();
+
                 foreach (Generated.CVParamType cv in _mzMLConnection.run.spectrumList.spectrum[oneBasedIndex - 1].precursorList.precursor[0].activation.cvParam)
                 {
-                    dissociationDictionary.TryGetValue(cv.accession, out dissociationType);
+                    if (dissociationDictionary.TryGetValue(cv.accession, out var scanDissociationType))
+                    {
+                        scanDissociationTypes.Add(scanDissociationType);
+                    }
+                }
+
+                if (scanDissociationTypes.Contains(DissociationType.ETD) && scanDissociationTypes.Contains(DissociationType.HCD))
+                {
+                    dissociationType = DissociationType.EThcD;
+                }
+                else if (scanDissociationTypes.Any())
+                {
+                    dissociationType = scanDissociationTypes.First();
```

Comment:
I think this is fine but it seems a little klugey.  I don't want to recommend a change but i feel like we could always allow a list of dissociation types here. For now ETHCD is fine b/c it is well known. But, int he future, mabye we call it ETD/HCD and then we can just do any combination ETC/ECD, ETC/CID HCD/CID.  etc. etc.